x-pack/plugin/core/src/main/resources/ecs-template.yml

```
@@ -0,0 +1,225 @@
+    # As most fields are keyword, these will be automatically be matched correctly. Only core fields were taken into account.
+    # DISCUSS: Should @timestamp and data_stream.* also be moved under dynamic templates?
+    dynamic_templates:
+
+      ### Index mappings
+
+      # Message field is always index
+      - message:
+          path_match: message
+          mapping:
+            type: match_only_text
+
+      ### Runtime mappings
+
+      # Matching all string fields by default to keyword
+      # TODO: Does first / last match win in the list?
+      - strings_as_keyword:
+          match_mapping_type: string
+          runtime:
+            type: keyword
+            # Do we need this for runtime?
+            ignore_above: 1024
+
+      # Matching all long fields
+      # This matches the following fields: bytes, packets, port, duration, severity, pid
+      # No path matching is needed as it is assumed these fields are already in the json doc a long
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "7" as string comes in, will it still be mapped correctly?
+      - long_as_long:
+          match_mapping_type: long
+          runtime:
+            type: long
+
+      # Matching all long and double fields
+      # This matches the following fields: event.risk_score
+      # No path matching is needed as it is assumed these fields are already in the json doc a float or double
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "7.7" as string comes in, will it still be mapped correctly?
+      - float_as_double:
+          match_mapping_type: float
+          runtime:
+            type: double
+      - double_as_double:
+          match_mapping_type: double
+          runtime:
+            type: double
+
+      # Matching all boolean fields
+      # This matches the following fields: exists
+      # No path matching is needed as it is assumed these fields are already in the json doc as boolean
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "true" as string comes in, will it still be mapped correctly?
+      - strings_as_boolean:
+          match_mapping_type: boolean
+          runtime:
+            type: boolean
+
+      # Matching all string fields by default to keyword
+      # TODO: Does first / last match win in the list?
+      - strings_as_keyword:
+          match_mapping_type: string
+          runtime:
+            type: keyword
+            # Do we need this for runtime?
+            ignore_above: 1024
+
+      # Any message field except the top level one is matched as text but not indexed
+      # TODO: Does this work? Which one has priority?
+      - match_message:
+          match_mapping_type: string
+          match: message
```

If you only want the top-level message field to be indexed, you can use

```suggestion
          path_match: message
```

---

I switched over to path_match and also removed match_mapping_type: string as I don't think it is needed in this case.